compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/phi-type-inference-property-store.expect.md

```diff
@@ -41,27 +41,22 @@ function Component(props) {
     t0 = $[0];
   }
   const x = t0;
-  let y;
+  let t1;
   if ($[1] !== props) {
+    let y;
     if (props.cond) {
       y = {};
     } else {
       y = { a: props.a };
     }
 
     y.x = x;
-    $[1] = props;
-    $[2] = y;
-  } else {
-    y = $[2];
-  }
-  let t1;
-  if ($[3] !== y) {
+
     t1 = [x, y];
```

Comment:
We now know the type of `y` and that type always invalidates, which makes these two scopes eligible for merging.